### PR TITLE
Feature: Add Drawer Menu in Home Screen

### PR DESCRIPTION
**Pull request by narendrasendhav**
- Added Drawer menu in home screen.
- Also added an icon-button in the home screen to open Drawer. Please check screenshots.

After adding drawer (**Notice icon button in home screen**):

home_screen_now
drawer

Before adding drawer:

home_screen_before

### CODE REVIEW

File: lib/modules/home/home_page.dart [hunk 1]
```diff
@@ -18,11 +19,19 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends ModularState<HomePage, HomeController> {
+    final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
```
Comment (7flash) [inline lib/modules/home/home_page.dart:22]: Move this to home_page_controller.dart

File: lib/modules/home/home_page.dart [hunk 2]
```diff
@@ -66,7 +75,14 @@ class _HomePageState extends ModularState<HomePage, HomeController> {
                   ),
                 ],
               ),
-            )
+            ),
+            Positioned(
+                top: 40,
+                left: 10,
+                child: IconButton(
+                  icon: Icon(Icons.dehaze),
```
Comment (7flash) [inline lib/modules/home/home_page.dart:83]: Use customized icon asset according to design

File: lib/modules/home/widgets/drawer_button.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    Key key,
+    this.iconPath = '',
+    this.title = '',
+    this.onTap,
+    this.isLogout = false,
+  }) : super(key: key);
+
+  final String iconPath, title;
+  final Function onTap;
+  final bool isLogout;
```
Comment (7flash) [inline lib/modules/home/widgets/drawer_button.dart:16]: make separate widget for logout

File: lib/modules/home/widgets/drawer_widget.dart [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                  ),
+                ],
+              ),
+            ),
+            DrawerButton(
+              iconPath: 'assets/ic_logout.svg',
+              title: Strings.logout,
+              isLogout: true,
+              onTap: () {
+                _closeDrawer();
```
Comment (7flash) [inline lib/modules/home/widgets/drawer_widget.dart:121]: implement logout function inside of controller and call controller.logout() here

```
  @action
  Future<void> logout() async {
    await auth.logout();
    await currentUserStore.setCurrentUser(null);
  }
```

File: lib/values/Strings.dart [hunk 5]
```diff
@@ -63,4 +63,13 @@ class Strings {
   ///Messages
   static String emailToChangePassword(String email) =>
       'An e-mail to change your password has been send to $email';
+
+  static const String dummyName = 'Jayden Massey';
```
Comment (7flash) [inline lib/values/Strings.dart:67]: remove this as being unused according to change above

File: lib/modules/home/widgets/drawer_widget.dart [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                ],
+              ),
+            ),
+            const SizedBox(
+              height: 28,
+            ),
+            Expanded(
+              child: Column(
+                children: [
+                  DrawerButton(
```
Comment (7flash) [inline lib/modules/home/widgets/drawer_widget.dart:71]: At this moment we need following buttons, and other will be added later.

My Profile
My Wallet
Activity History
Bookmarks

all four of them should navigate to profile at the moment

File: lib/modules/home/widgets/drawer_widget.dart [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+                  ),
+                  Text(
+                    Strings.dummyName,
+                    style: bold16White,
+                  ),
+                  const SizedBox(
+                    height: 8,
+                  ),
+                  Text(
+                    Strings.dummyName,
```
Comment (7flash) [inline lib/modules/home/widgets/drawer_widget.dart:59]: should show controller.currentUser.status

File: lib/modules/home/widgets/drawer_widget.dart [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+                        radius: 26,
+                        backgroundImage: AssetImage('assets/ic_avatar.jpg'),
+                      ),
+                    ),
+                  ),
+                  const SizedBox(
+                    height: 8,
+                  ),
+                  Text(
+                    Strings.dummyName,
```
Comment (7flash) [inline lib/modules/home/widgets/drawer_widget.dart:52]: show controller.currentUser.name

File: lib/modules/home/home_page.dart (hunk 1 above)
Comment (narendrasendhav) [reply lib/modules/home/home_page.dart:22]: Okay, I will move it to home_page_controller.dart.

File: lib/modules/home/home_page.dart (hunk 2 above)
Comment (narendrasendhav) [reply lib/modules/home/home_page.dart:83]: Okay, sure.

File: lib/modules/home/widgets/drawer_widget.dart (hunk 4 above)
Comment (narendrasendhav) [reply lib/modules/home/widgets/drawer_widget.dart:121]: Thanks for the logout function. I will use it.

File: lib/values/Strings.dart (hunk 5 above)
Comment (narendrasendhav) [reply lib/values/Strings.dart:67]: Okay, I will remove it.

File: lib/modules/home/widgets/drawer_widget.dart (hunk 6 above)
Comment (narendrasendhav) [reply lib/modules/home/widgets/drawer_widget.dart:71]: Okay, I will navigate all of these buttons to profile screen.

File: lib/modules/home/widgets/drawer_widget.dart (hunk 7 above)
Comment (narendrasendhav) [reply lib/modules/home/widgets/drawer_widget.dart:59]: Okay, thanks. I will update it.

File: lib/modules/home/widgets/drawer_widget.dart (hunk 8 above)
Comment (narendrasendhav) [reply lib/modules/home/widgets/drawer_widget.dart:52]: Okay, thanks. I will update it.

File: lib/modules/home/widgets/drawer_widget.dart (hunk 6 above)
Comment (narendrasendhav) [reply lib/modules/home/widgets/drawer_widget.dart:71]: Hi Igor, I had some doubts regarding this change. Out of the buttons that you mentioned above, there are only profile avatar image, wallet button & activity history button in drawer menu. Should I remove the remaining buttons? Please let me know which icon to use for "Bookmarks" button?

Please clarify if "My Profile" button that you are referring to, is the Profile avatar?

Comment (7flash) [reply lib/modules/home/widgets/drawer_widget.dart:71]: Please find the most appropriate icon for Bookmarks, and remove all other buttons but create these: My Profile, My Wallet, Activity History, Bookmarks

Comment (narendrasendhav) [reply lib/modules/home/widgets/drawer_widget.dart:71]: Igor, I have added the buttons - My Profile, My Wallet, Activity History, Bookmarks. But the app is crashing when navigating to Profile screen & also on logout button press.

Profile screen crash:
Screenshot_2020-06-08-09-31-51-285_network highvibe hvn

Logout out button crash:
Screenshot_2020-06-08-09-37-42-748_network highvibe hvn

**Pull request merged**